modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -470,10 +477,16 @@ RequestMetricCollector getMetricCollector(Operation operation, OperationPurpose
             return collectors.computeIfAbsent(new StatsKey(operation, purpose), k -> buildMetricCollector(k.operation(), k.purpose()));
         }
 
-        Map<String, Long> statsMap() {
-            final Map<String, Long> m = Arrays.stream(Operation.values()).collect(Collectors.toMap(Operation::getKey, e -> 0L));
-            collectors.forEach((sk, v) -> m.compute(sk.operation().getKey(), (k, c) -> Objects.requireNonNull(c) + v.counter.sum()));
-            return Map.copyOf(m);
+        Map<String, Long> statsMap(boolean isStateless) {
+            if (isStateless) {
```

Comment:
This same method is also used for [public facing API](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-repositories-metering-api.html). So we need it preverse its behaviour for non-stateless. There are different ways to achieve this. But I think passing the flag here is likely the simplest. And given the nature of this work, it does not feel necessary to go with a more elaborative approach. I hope this makes sense.